Content.Server/GridPreloader/GridPreloaderSystem.cs

```diff
@@ -0,0 +1,101 @@
+using Content.Server.GameTicking.Events;
+using Content.Shared.GridPreloader.Prototypes;
+using Content.Shared.GridPreloader.Systems;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Prototypes;
+using System.Linq;
+using System.Numerics;
+
+namespace Content.Server.GridPreloader;
+public sealed partial class GridPreloaderSystem : SharedGridPreloaderSystem
+{
+
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly MapLoaderSystem _map = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+
+        SubscribeLocalEvent<RoundStartingEvent>(OnRoundStart);
+    }
+
+    private void OnRoundStart(RoundStartingEvent ev)
+    {
+        //Init preloader entity in nullspace
+        var preloaderEntity = Spawn();
+        var preloader = AddComp<GridPreloaderComponent>(preloaderEntity);
+
+        preloader.PreloadGridsMapId = _mapManager.CreateMap();
+        _mapManager.AddUninitializedMap(preloader.PreloadGridsMapId);
+        _mapManager.SetMapPaused(preloader.PreloadGridsMapId, true);
+
+        var globalXOffset = 0f;
+        foreach (var proto in _prototype.EnumeratePrototypes<PreloadedGridPrototype>())
+        {
+            for (int i = 0; i < proto.Copies; i++)
+            {
+                var options = new MapLoadOptions
+                {
+                    LoadMap = false,
+                };
+
+                // i dont use TryLoad, because he doesn't return EntityUid
+                var gridUid = _map.LoadGrid(preloader.PreloadGridsMapId, proto.Path.ToString(), options);
+
+                if (gridUid == null)
+                    continue;
+
+                EnsureComp<MapGridComponent>(gridUid.Value, out var mapGrid);
+                EnsureComp<PhysicsComponent>(gridUid.Value, out var physic);
+
+                //Position Calculating
+                globalXOffset += mapGrid.LocalAABB.Width / 2;
+
+                var xPos = -physic.LocalCenter.X + globalXOffset;
+                var yPos = -physic.LocalCenter.Y;
+
+                _transform.SetLocalPosition(Transform(gridUid.Value), new Vector2(xPos, yPos));
+                _transform.SetLocalRotation(Transform(gridUid.Value), Angle.Zero);
+
+                globalXOffset += (mapGrid.LocalAABB.Width / 2) + 1;
+
+                //Add to list
+                preloader.PreloadedGrids.Add(new(proto.ID, gridUid.Value));
```

Comment:
1. You're resolving transformcomponent twice which is expensive.
2. Don't break out the vector to 2 fields it's unnecessary.
3. Use SetCoordinates as what's here will be raising 2 moveevents instead of 1 (also expensive).
4. Rather than bumping the offset twice just bump it once at the start with the width or change the first one to half the length of the prior one (preferably the format).